main.go

```diff
@@ -277,6 +280,14 @@ func main() {
 		return nil
 	}))
 
+	if err = (&appmeshcontroller.SidecarReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("Sidecar"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "Sidecar")
+		os.Exit(1)
+	}
```

Comment:
this stuff would be wrapped into a "NewSidecarReconciler" so not needed here but scaffolding injected it here